test/tests/interfaces/pbs_queue_partition.py

```diff
@@ -0,0 +1,209 @@
+# details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# The PBS Pro software is licensed under the terms of the GNU Affero General
+# Public License agreement ("AGPL"), except where a separate commercial license
+# agreement for PBS Pro version 14 or later has been executed in writing with
+# Altair.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software - under
+# a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.interfaces import *
+
+
+@tags('multisched')
+class TestQueuePartition(TestInterfaces):
+
+    """
+    Test suite to test partition attr for queue
+    """
+
+    def create_queue_partition(
+            self,
+            q_type="e",
+            q_name="Q1",
+            partition="P1",
+            enable="True",
+            start="True"):
+        """
+        Common function to create queue with partition
+        :param q_type: "queue_type" attribute of queue object ,
+                    Defaults to "execution"
+        :type q_type:  str
+        :param q_name: name of the queue, Defaults to "Q1"
+        :type q_name: str
+        :param partition: "partition" attribute of queue object,
+                    Defaults to "P1"
+        :type partition: str
+        :param enable: "enabled" attribute of queue object, Defaults to "True"
+        :type enable: boolean
+        :param start: "started" attribute of queue object, Defaults to "True"
+        :type start: boolean
+        """
+        attr = {
+            'queue_type': q_type,
+            'enabled': enable,
+            'started': start,
+            'partition': partition}
+        self.server.manager(MGR_CMD_CREATE,
+                            QUEUE, attr,
+                            id=q_name, expect=True)
+
+    def test_set_unset_partition(self):
+        """
+        Test to set/unset the partition attribute of queue object
+        """
+        self.create_queue_partition()
+        self.server.manager(MGR_CMD_SET,
+                            QUEUE, {'partition': "P2"},
+                            id="Q1", expect=True)
+        # resetting the same partition value
+        self.server.manager(MGR_CMD_SET,
+                            QUEUE, {'partition': "P2"},
+                            id="Q1", expect=True)
+        self.server.manager(MGR_CMD_UNSET,
+                            QUEUE, 'partition',
+                            id="Q1", expect=True)
+
+    def test_set_partition_user_permissions(self):
+        """
```

Comment:
From @suresh-thelkar analysis we can  do some fix in test case. I suggest to have fix in framework side return errorCode in rc.  @kjakkali Please take a look into thread. 

Continuing  @suresh-thelkar analysis. 
rc can have two values   (errorCode) or   (errorCode % 256)
In windows(qmgr CLI)   -  rc  =  (errorCode) -   15217
In Api mode                     -  rc  = (errorCode)  -  15217
In Linux (qmgr CLI)        -  rc  = (errorCode%256) - 113


Now , what will happen when we always assert (errorCode%256)
something like 
      self.assertTrue( (rc%256) == (errorCode%256), "Fail")
------
In windows(qmgr CLI)   -        self.assertTrue( (15217%256) == (15217%256), "Fail")  - self.assertTrue( 113 == 113, "Fail")
In Api mode                     -        self.assertTrue( (15217%256) == (15217%256), "Fail") - self.assertTrue( 113 == 113, "Fail")
In Linux (qmgr CLI)        -        self.assertTrue( (113%256) == (15217%256), "Fail") - self.assertTrue( 113 == 113, "Fail")

So all test cases are protected.  Solves all problems